ArduPlane/tailsitter.cpp

```diff
@@ -267,19 +275,29 @@ bool QuadPlane::tailsitter_transition_vtol_complete(void) const
 {
     if (plane.fly_inverted()) {
         // transition immediately
+        gcs().send_text(MAV_SEVERITY_INFO, "Transition VTOL done, inverted flight");
         return true;
     }
     // for vectored tailsitters at zero pilot throttle
-    if ((plane.quadplane.get_pilot_throttle() < .05f) && plane.quadplane._is_vectored) {
+    if (plane.control_mode->is_vtol_man_throttle() && (plane.quadplane.get_pilot_throttle() < .05f) && plane.quadplane._is_vectored) {
```

Comment:
what does this change do?

Reply:
It means we only check the pilot throttle if its a pilot controlled mode. Currently you could be loitering in AUTO with 0 pilot throttle and it would do a instant transition.

Reply:
We could change to ask attitude control for the throttle rather than pilot, then should work for all cases

Reply:
does this interact with @Hwurzburg 's changes to allow landing in FW modes with the motors tilted up? 
I don't guess it should unless this is used to manually override motor tilt during an unplanned auto mode landing.

Reply:
does this revert the prop strikes on the ground when switching from a fixed wing mode like fbwa to qloiter while armed?

Reply:
TBH I'm not really a fan of this if at all. Currently it means a vectored tailsitter will transition to VTOL instantly if there is low throttle and low ground speed. I'm not sure why transitioning faster prevents prop strikes?

Now we have @Hwurzburg 's force flair switch option to tilt up in forward flight I think this is redundant and should be removed.

Reply:
its a real problem...you land in FBWA at zero throttle and while armed switch to any VTOL mode to takeoff again you will prop strike....forcing someone to use a switch to implement the flare action does not seem reasonable to prevent prop strikes...perhaps an additional qualifier such as !is_flying() should be added?

Reply:
So the issue is that the props spin up for the transition, but the tilts are not moved to point up yet? This is not really a fix for that, the props will still spin up but the servos cant react so fast you will still prop strike.

I think the correct fix is to not spin up the props in the first place. Maybe we could look at the throttle suppression code....

Reply:
it is a lot better than sitting there trying to spin for 1 sec before moving...the props clear the ground fast enough to not stall the esc even as they ramp up....some kind of throttle wait period is involved also...but if we can stop the spin_arm, spin_min stuff on tailsitters until the throttle is moved < 5% that would work also I think...but that throttle wait period has to go also I would think...

Reply:
maybe we should leave as it for this PR and propose a different fix for the prop strike/loiter no ground speed issue later? really scope creep here

Reply:
Fair comment, I have dropped it from this PR. I had a play in RF I see the issue but I think the fix should be on this line https://github.com/ArduPilot/ardupilot/blob/b8d4dbfbb3d71a7b7282f4558af4b9078d75ee2d/ArduPlane/tailsitter.cpp#L80
That would disable the throttle for the transition rather than skipping the transition completely. I did have a look at the plane throttle suppression flag, but it would not do this, I think `get_pilot_throttle()` is probably the best hope, so long as its called in modes where the pilot is actually directly in control of the throttle.